docs/recipes/upgrading/v4-to-v5.md

```diff
@@ -0,0 +1,257 @@
+
+	- You can run to upgrade all at once `npx npm-check-updates -u` or if you prefer to be more conservative you can use `npx npm-check-updates -i`.
+	
+	If you use the second option, make sure to select `react`, `react-dom`, `sass`.
+	
+1. Install the new NPM dependencies
+
+	`npm install -D vite @vitejs/plugin-react tailwindcss postcss autoprefixer @types/node`
+	
+1. Delete `webpack.config.js`
+
+1. Create the file `src/Client/vite.config.mts` with the following content
+	
+	```ts
+	import { defineConfig } from "vite";
+	import react from "@vitejs/plugin-react";
+	
+	const proxyPort = process.env.SERVER_PROXY_PORT || "5000";
+	const proxyTarget = "http://localhost:" + proxyPort;
+	
+	// https://vitejs.dev/config/
+	export default defineConfig({
+	    plugins: [react()],
+	    build: {
+	        outDir: "../../deploy/public",
+	    },
+	    server: {
+	        port: 8080,
+	        proxy: {
+	            // redirect requests that start with /api/ to the server on port 5000
+	            "/api/": {
+	                target: proxyTarget,
+	                changeOrigin: true,
+	            }
+	        }
+	    }
+	});
+	```
+
+1. Overwrite `src/Client/index.html` with the following content
+	
+	```html
+	<!doctype html>
+	<html>
+	<head>
+	    <title>SAFE Template</title>
+	    <meta charset="utf-8">
+	    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+	    <link rel="icon" type="image/png" href="/favicon.png"/>
+	    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.14.0/css/all.min.css">
+	</head>
+	<body>
+	    <div id="elmish-app"></div>
+	    <!-- Polyfill for remotedev -->
+	    <script type="text/javascript">
+	        var global = global || window;
+	    </script>
+	    <script type="module" src="/output/App.js"></script>
+	</body>
+	</html>
+	```
+	
+1. Run `npx tailwindcss init -p` in `src/Client`
+1. Change `tailwind.config.js` content with
+
+	```js
+	/** @type {import('tailwindcss').Config} */
+	module.exports = {
+	    mode: "jit",
+	    content: [
+	        "./index.html",
+	        "./**/*.{fs,js,ts,jsx,tsx}",
+	    ],
+	    theme: {
+	        extend: {},
+	    },
+	    plugins: []
+	}
+	```
+
+1. In order to get VSCode Tailwind intellisense, user needs to install `Tailwind CSS Intellisense`. We also create the file `.vscode/settings.json` with the following content.
+
+	*The regex here are for Feliz style DSL, if you want to support Fable.React DSL you need to adapt the regexes*
+	
+	```json
+	{
+	    "tailwindCSS.includeLanguages": {
+	        "fsharp": "html"
+	    },
+	    "tailwindCSS.experimental.classRegex": [
+	        "\\.className\\s+\"([^\"]*)\"",
+	        ["\\.className\\s+\\[([^\\]]*)\\]", "\"([^\"]*)\""]
+	    ]
+	}
+	```
+
+1. Create a file `src/Client/index.css` with the following content
+
+	```css
+	@tailwind base;
+	@tailwind components;
+	@tailwind utilities;
+	```
+	
+1. Add the following lines at the top of `src/Client/App.fs`
```

Comment:
Presumably after the module declaration? We should be more precise with wording here.

Reply:
The tidiest place to put it would be after the other open declarations, so that we're not mixing open declarations and codes

Reply:
Mention directly after open Elmish.React

Reply:
This is unresolved.